tests/robustness/engine/action.go

```diff
@@ -111,8 +140,8 @@ func (e *Engine) checkErrRecovery(ctx context.Context, incomingErr error, action
 			return outgoingErr
 		}
 
-		e.RunStats.IncrementDataPurgeCount()
-		e.CumulativeStats.IncrementDataPurgeCount()
+		e.RunStats.DataPurgeCount++
+		e.CumulativeStats.DataPurgeCount++
```

Comment:
Suggestion: Put these stats related counter updates into subroutine which accepts some sort of discriminator. Then one can safely protect them under all circumstances. e.g.
```
func (e *Engine) statsUpdateCounters(updateType string) {
	e.statsMux.Lock()
	defer e.statsMux.Unlock()
	switch updateType {
         case "incr-no-op": ...
         case "incr-data-purge":  ...
         case "incr-data-restore": ... 
         case "incr-error-recovery": ...
        }
}
```